axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/ProjectTaskReportingValuesComputingServiceImpl.java

```diff
@@ -0,0 +1,395 @@
+package com.axelor.apps.businessproject.service;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Product;
+import com.axelor.apps.base.db.Unit;
+import com.axelor.apps.base.db.repo.TraceBackRepository;
+import com.axelor.apps.businessproject.exception.BusinessProjectExceptionMessage;
+import com.axelor.apps.businessproject.service.app.AppBusinessProjectService;
+import com.axelor.apps.hr.db.TimesheetLine;
+import com.axelor.apps.hr.db.repo.EmployeeRepository;
+import com.axelor.apps.hr.db.repo.TimesheetLineRepository;
+import com.axelor.apps.hr.db.repo.TimesheetRepository;
+import com.axelor.apps.hr.exception.HumanResourceExceptionMessage;
+import com.axelor.apps.project.db.Project;
+import com.axelor.apps.project.db.ProjectPlanningTime;
+import com.axelor.apps.project.db.ProjectTask;
+import com.axelor.apps.project.db.repo.ProjectRepository;
+import com.axelor.apps.project.db.repo.ProjectTaskRepository;
+import com.axelor.apps.sale.db.SaleOrderLine;
+import com.axelor.i18n.I18n;
+import com.axelor.studio.db.AppBusinessProject;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.Objects;
+
+public class ProjectTaskReportingValuesComputingServiceImpl
+    implements ProjectTaskReportingValuesComputingService {
+
+  private ProjectTaskRepository projectTaskRepo;
+  private TimesheetLineRepository timesheetLineRepository;
+  private AppBusinessProjectService appBusinessProjectService;
+
+  public static final int RESULT_SCALE = 2;
+  public static final int COMPUTATION_SCALE = 5;
+
+  // AppBusinessProject config
+  private Unit daysUnit;
+  private Unit hoursUnit;
+  private BigDecimal defaultHoursADay;
+
+  @Inject
+  public ProjectTaskReportingValuesComputingServiceImpl(
+      ProjectTaskRepository projectTaskRepo,
+      TimesheetLineRepository timesheetLineRepository,
+      AppBusinessProjectService appBusinessProjectService) {
+    this.projectTaskRepo = projectTaskRepo;
+    this.timesheetLineRepository = timesheetLineRepository;
+    this.appBusinessProjectService = appBusinessProjectService;
+  }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void computeProjectTaskTotals(ProjectTask projectTask) throws AxelorException {
+
+    // get AppBusinessProject config
+    AppBusinessProject appBusinessProject = appBusinessProjectService.getAppBusinessProject();
+    daysUnit = appBusinessProject.getDaysUnit();
+    hoursUnit = appBusinessProject.getHoursUnit();
+    defaultHoursADay = appBusinessProject.getDefaultHoursADay();
```

Comment:
please fetch these fields using a dedicated method in AppBusinessProjectService that throws an AxelorException if the configuration is empty, to warn the user that the config must be filled